File: spring-springmvc-mybatis/src/main/java/com/jiangKlijna/web/service/impl/UserServiceImpl.java
package com.jiangKlijna.web.service.impl;

import com.jiangKlijna.web.dao.IUserDao;
import org.springframework.stereotype.Service;

import com.jiangKlijna.web.bean.Result;
import com.jiangKlijna.web.bean.User;
import com.jiangKlijna.web.service.BaseService;
import com.jiangKlijna.web.service.UserService;

import javax.annotation.Resource;

@Service("userService")
public class UserServiceImpl extends BaseService implements UserService {

    @Resource
    private IUserDao userDao;

    @Override
    public Result regist(String username, String password) {
        try {
            userDao.save(new User(username, password));
            return sucessResult();
        } catch (Exception e) {
            return errorResult(e.toString());
        }
    }

    @Override
    public Result remove(int id) {
        try {
            userDao.delete(id);
            return sucessResult();
        } catch (Exception e) {
            return errorResult(e.toString());
        }
    }

    @Override
    public Result find(int id) {
        try {
            return sucessResult(userDao.get(id));
        } catch (Exception e) {
            return errorResult(e.toString());
        }
    }

}

File: spring-springmvc-mybatis-kotlin/src/main/java/com/jiangKlijna/web/service/impl/UserServiceImpl.kt
package com.jiangKlijna.web.service.impl

import com.jiangKlijna.web.dao.IUserDao
import org.springframework.stereotype.Service

import com.jiangKlijna.web.bean.Result
import com.jiangKlijna.web.bean.User
import com.jiangKlijna.web.service.BaseService
import com.jiangKlijna.web.service.UserService

import javax.annotation.Resource

@Service("userService")
class UserServiceImpl : BaseService(), UserService {

    @Resource
    private val userDao: IUserDao? = null

    override fun regist(username: String, password: String): Result {
        try {
            userDao!!.save(User(username, password))
            return sucessResult()
        } catch (e: Exception) {
            return errorResult(e.toString())
        }

    }

    override fun remove(id: Int): Result {
        try {
            userDao!!.delete(id)
            return sucessResult()
        } catch (e: Exception) {
            return errorResult(e.toString())
        }

    }

    override fun find(id: Int): Result {
        try {
            return sucessResult(userDao!!.get(id))
        } catch (e: Exception) {
            return errorResult(e.toString())
        }

    }

}
